Modules/_ctypes/callproc.c

```diff
@@ -2022,7 +2022,7 @@ create_pointer_inst(PyObject *module, PyObject *arg)
         return NULL;
     }
     if (typ == NULL) {
-        typ = create_pointer_type(NULL, (PyObject *)Py_TYPE(arg));
```

Comment:
This change is not needed here; so I'll comment as if it *was* needed. Feel free to leave it to future commit (but let's not forget it):

Could you also add an `assert(module)` to `create_pointer_type`?
(And in general, to any function that starts needing a previously unused argument?)

Reply:
Addressed in gh-117181: 28696c2e33ed1e690d2c29ed436b45603b434389